src/backend/joanie/tests/core/api/admin/course_product_relations/test_generate_certificates.py

```diff
@@ -0,0 +1,770 @@
+            ),
+        )
+
+        response = self.client.delete(
+            f"/api/v1.0/admin/course-product-relations/{cpr.id}/generate_certificates/",
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.METHOD_NOT_ALLOWED)
+
+    def test_admin_api_course_product_relation_generate_certificates_authenticated_create_status(
+        self,
+    ):
+        """
+        Authenticated staff user should be able to trigger the generation of certificate if the
+        course's product is eligible. We should get a status code 201 CREATED in return
+        if all certificates were generated. When the task has been successful, the cache data
+        is deleted.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+
+        course = factories.CourseFactory(products=None)
+        product = factories.ProductFactory(
+            price="0.00",
+            type=enums.PRODUCT_TYPE_CREDENTIAL,
+            certificate_definition=factories.CertificateDefinitionFactory(),
+            courses=[course],
+        )
+        factories.CourseRunFactory(
+            course=course,
+            enrollment_end=timezone.now() + datetime.timedelta(hours=1),
+            enrollment_start=timezone.now() - datetime.timedelta(hours=1),
+            is_gradable=True,
+            start=timezone.now() - datetime.timedelta(hours=1),
+        )
+        factories.ProductTargetCourseRelationFactory(
+            product=product,
+            course=course,
+            is_graded=True,
+        )
+        cpr = CourseProductRelation.objects.get(product=product, course=course)
+        orders = factories.OrderFactory.create_batch(
+            10,
+            product=cpr.product,
+            course=cpr.course,
+        )
+        for order in orders:
+            order.submit()
+
+        self.assertFalse(Certificate.objects.exists())
+
+        with mock.patch.object(DummyLMSBackend, "get_grades") as mock_get_grades:
+            mock_get_grades.return_value = {"passed": True}
+
+            response = self.client.post(
+                f"/api/v1.0/admin/course-product-relations/{cpr.id}/generate_certificates/",
+                content_type="application/json",
+            )
+
+        self.assertEqual(response.status_code, HTTPStatus.CREATED)
+        self.assertDictEqual(
+            response.json(),
+            {
+                "in_progress": True,
+                "course_product_relation_id": str(cpr.id),
+                "count_certificate_to_generate": len(orders),
+                "count_exist_before_generation": 0,
+            },
+        )
+
+        for order in orders:
+            order.refresh_from_db()
+            certificate = Certificate.objects.get(order=order)
+            self.assertTrue(certificate)
```

Comment:
How a `Certificate` object can be `True` ? You should probably use a better assertion here. If you just want to check if the record exists, then you should use `exists()` with your queryset !